src/engraving/compat/midi/compatmidirenderinternal.cpp

```diff
@@ -631,8 +631,8 @@ static int calculateTieLength(const Note* note)
     const Note* n = note;
     while (n) {
         // Process ties or bends
-        Tie* tieFor = n->tieFor();
-        GuitarBend* bendFor = n->bendFor();
+        const Tie* tieFor = n->tieFor();
```

Comment:
```suggestion
        const Tie* tieFor = n->tieForNonPartial();
```
will only return normal ties, excluding partials and laissez vibs which definitely won't have end notes.